rest/adminapitest/admin_api_test.go

```diff
@@ -1433,6 +1433,110 @@ func TestResyncStop(t *testing.T) {
 	assert.True(t, syncFnCount < 2000, "Expected syncFnCount < 2000 but syncFnCount=%d", syncFnCount)
 }
 
+// TestCorruptDbConfigHandling:
+//   - Create persistent config rest tester
+//   - Create a db with a non-corrupt config
+//   - Grab the persisted config from the bucket for purposes of changing the bucket name on it (this simulates a
+//     dbconfig becoming corrupt)
+//   - Update the persisted config to change the bucket name to a non-existent bucket
+//   - Assert that the db context and config are removed from server context and that operations on the database GET and
+//     DELETE fail with appropriate error message for user
+//   - Test we are able to update the config to correct the corrupted db config
+//   - asser the db returns to the server context and is removed from corrupt database tracking AND that the bucket name
+//     on the config now matches the rest tester bucket name
+func TestCorruptDbConfigHandling(t *testing.T) {
+	base.LongRunningTest(t)
+	if base.TestsRequireBootstrapConnection() {
+		t.Skip("Walrus/Rosmar does not support persistent config")
+	}
+	numCollections := 1
+	base.RequireNumTestDataStores(t, numCollections)
```

Comment:
there should always be one collection - if it requires a named collection, you should test for that - use `SG_TEST_USE_DEFAULT_COLLECTION=true` and `SG_TEST_USE_GSI=false` permutations.